communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java

```diff
@@ -80,54 +80,50 @@ public GAVDependencyTree readRelationships(File pomRepoDir, File pomPath,
                 Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
                         findAllPomFiles(pomRepoDir));
 
-                ProjectVersionRef rootProject = null;
-
                 // for each pom.xml
                 for (Map.Entry<File, ProjectVersionRef> entry : projectVersionRefs.entrySet()) {
-                    // if pomPath is the same as what we found analysis the root repository
-                    if (pomPath.getCanonicalPath().equals(entry.getKey().getCanonicalPath())) {
-                        rootProject = entry.getValue();
-                        break;
+                    Set<ProjectRelationship<?, ?>> relationships;
+                    try {
+                        // find all the dependencies / plugins / parents / plugins etc for each pom.xml
+                        relationships = getDiscoveryResult(tempDir, pomRepoDir, entry.getValue(),
+                                repositories).getAcceptedRelationships();
```

Comment:
@jbartece Maybe I wrongly understood what you meant!
Do we actually want to see only the top-level, or the full dependency tree?